app/models/concerns/teacher.rb

```diff
@@ -139,4 +141,54 @@ def update_teacher params
     end
     response
   end
-end
\ No newline at end of file
+
+  def is_premium?
+    subscriptions
+      .where("subscriptions.expiration >= ?", Date.today)
+      .where("subscriptions.account_limit >= ?", self.my_students.length)
+      .any?
+  end
+
+  def my_students
```

Comment:
I think this is redundant, because teacher.students is already set up as an association.